source/common/router/retry_state_impl.cc

```diff
@@ -225,6 +225,10 @@ RetryStatus RetryStateImpl::shouldRetry(bool would_retry, DoRetryCallback callba
   }
 
   if (retries_remaining_ == 0) {
+    cluster_.stats().upstream_rq_retry_limit_exceeded_.inc();
```

Comment:
If there is no retry policy at all does this increment? Mostly just trying to figure out how valuable this stat is operationally?

Reply:
No, it would not increment if there's no policy at all. Calls into this function would come in with `would_retry == false` and we would return above without incrementing.